pytket/docs/extensions.md

```diff
@@ -0,0 +1,131 @@
+
+The additional modules can be installed adding the extension name to the installation command for pytket. For example pytket-quantinuum can be installed by running
+
+```
+pip install pytket-quantinuum
+```
+
+The types of `Backend` available in pytket are the following
+
+## Types of Backend
+
+- **QPUs** - These are real quantum computers that return shots based results. E.g the [QuantinuumBackend](inv:#*.QuantinuumBackend).
+- **Cloud Access** - Cloud backends allow pytket to interface with cloud platforms to access additional QPUs and simulators. E.g [BraketBackend](inv:#*.BraketBackend).
+- **Emulators** - These classically simulate a circuit and produce shots based results. Sometimes emulators use a noise model and have connectivity constraints to emulate real QPUs. E.g. [IBMQEmulatorBackend](inv:#*.IBMQEmulatorBackend)
+- **Statevector Simulators** - Calculates the pure quantum state prepared by a circuit returning a vector/ndarray. Examples of statevector simulators are the [ForestStateBackend](inv:#*.ForestStateBackend) and the [AerStateBackend](inv:#*.AerStateBackend).
+- **Unitary Simulators** - Unitary simulators calculate the unitary operator that is applied by a circuit. A unitary matrix/ndarray is returned [AerUnitaryBackend](inv:#*.AerUnitaryBackend) is an example of such a simulator.
+- **Density Matrix Simulators** - These simulators compute the density matrix prepared by a circuit. The result can be a statistical mixture of states in contrast to statevector simulation. E.g. [CirqDensityMatrixSampleBackend](inv:#*.CirqDensityMatrixSampleBackend)
+- **Other specialised simulators** - There are extensions for simulating specific types of circuit. For instance the [SimplexBackend](inv:#*.SimplexBackend) is designed to simulate Clifford circuits.
+
+A full list of available pytket backends is shown below.
+
+## QPUs
+
+[QuantinuumBackend](inv:#*.QuantinuumBackend)
+\- Interface to a remote Quantinuum device or simulator. There are currently two Quantinuum devices offered (H1-1 and H2-1).
+
+[IBMQBackend](inv:#*.IBMQBackend)
+\- A backend for running circuits on remote IBMQ devices.
+
+[ForestBackend](inv:#*.ForestBackend)
+\- A backend for running circuits on remote Rigetti devices.
+
+[AQTBackend](https://cqcl.github.io/pytket-aqt/api/api.html#pytket.extensions.aqt.AQTBackend)
+\- Interface to an AQT device or simulator.
+
+[IQMBackend](inv:#*.IQMBackend)
+\- Interface to an IQM device or simulator.
+
+## Cloud Access
+
+[BraketBackend](inv:#*.BraketBackend)
+\- Interface to Amazon Braket service.
+
+## Emulators
+
+[IBMQEmulatorBackend](inv:#*.IBMQEmulatorBackend) - A backend which uses the [AerBackend](inv:#*.AerBackend) to emulate the behavior of IBMQBackend.
+
+[QuantinuumBackend](inv:#*.QuantinuumBackend)
```

Comment:
Note that there are some additional build warnings given by this new syntax. I think these are due to the fact that `QuantinuumBackend` (and other backends) don't have a unique import path.

The links do work however and I've ran a link checker over these pages to fix any broken links.

I'll make an issue to fix the warnings. The output is quite cluttered already

Reply:
I can just see one warning that might be important, but it can be fixed in another PR (or this one if you prefer):
```
/home/runner/work/tket/tket/pytket/pytket/utils/operators.py:docstring of pytket.utils.QubitPauliOperator.all_qubits:7: WARNING: Definition list ends without a blank line; unexpected unindent.
```

Reply:
I'd rather fix that last warning in a follow up PR if thats okay. Not immediately sure what the matter is.